### PR TITLE
✨ Source Instagram: Migrate MediaInsights and StoryInsights to low-code

### PR DESCRIPTION
**Pull request by aldogonzalez8**
## What
This is the follow-up of the initial PR to [migrate source-instagram to low-code](https://github.com/airbytehq/airbyte/pull/39504/files), this one scope for:
- MediaInisghts
- StoryInisghts

Note: Also, there was [another PR](https://github.com/airbytehq/airbyte/pull/40524) to resolve an issue when fetching accounts endpoint and the response includes non-business accounts

Resolving [Source Instagram: Migrate to low code
](https://github.com/airbytehq/airbyte-internal-issues/issues/7909)
## How
MediaInisghts and StoryInsights are now included in the manifest, and the JSON schema was removed.

**Important:** I'm including new fields in Media and Stories to pass extra data to child streams.
- media_insights_info
- story_insights_info

If you can provide feedback about this pattern, I would appreciate it.

## Review guide

1. `source_instagram/manifest.yaml`: adding MediaInisghts, InstoryInsights and also media_insights_info field in Media and story_insights_info field in Stories
2. `source_instagram/components.py`: new custom component for insights transformations
3. `source_instagram/source.py`: remove import of old python sources that are being replaced
4. `source_instagram/streams.py`: remove old python sources that are being replaced
5. The rest is integration testing around both streams to ensure functionality, specially with MediaInsights

## User Impact
It should be a transparent migration for users

## Can this PR be safely reverted and rolled back?
- [x] YES 💚
- [ ] NO ❌

### CODE REVIEW

**Comment by aldogonzalez8** [comment]
Live tests have already been executed.

image

image

image

File: airbyte-integrations/connectors/source-instagram/source_instagram/manifest.yaml [hunk 1]
```diff
@@ -314,6 +438,77 @@ definitions:
...
+                max_retries: 5
+                backoff_strategies:
+                  - type: ExponentialBackoffStrategy
+                    factor: 5
+                response_filters:
+                  - type: HttpResponseFilter
+                    action: IGNORE
+                    predicate: >-
+                      {{ 'error' in response and response['error']['code'] == 10
+                      }}
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-instagram/source_instagram/manifest.yaml:465]: nit: format

File: airbyte-integrations/connectors/source-instagram/source_instagram/manifest.yaml [hunk 2]
```diff
@@ -314,6 +438,77 @@ definitions:
...
+                    error_message: >-
+                      Insights error: {{
+                      response['error']['message'] }}
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-instagram/source_instagram/manifest.yaml:468]: nit: format

**Review by bazarnov — APPROVED** [review]
Looks good to me! Nice work @aldogonzalez8

File: airbyte-integrations/connectors/source-instagram/source_instagram/manifest.yaml (hunk 1 above)
Comment (aldogonzalez8) [reply airbyte-integrations/connectors/source-instagram/source_instagram/manifest.yaml]: Fixed

File: airbyte-integrations/connectors/source-instagram/source_instagram/manifest.yaml (hunk 2 above)
Comment (aldogonzalez8) [reply airbyte-integrations/connectors/source-instagram/source_instagram/manifest.yaml]: Fixed

**Pull request merged**